aws/rakefiles/deploy.rake

```diff
@@ -17,20 +17,14 @@ desc "Wait until GPII components have been deployed"
 task :wait_for_gpii_ready => :configure_kubectl do
   puts "Waiting for GPII components to be fully deployed..."
   puts "(You can Ctrl-C out of this safely. You may need to re-run :deploy_only afterward.)"
-
-  # This is the simplest one-liner I could find to GET a url and return just
-  # the status code.
-  # http://superuser.com/questions/590099/can-i-make-curl-fail-with-an-exitcode-different-than-0-if-the-http-status-code-i
-  #
-  # The grep catches a 2xx status code.
-  #
-  # We use /preferences/carla as a proxy for the overall health of the system.
-  # It's not perfect but it's a good start.
   preferences_url = "https://preferences.#{ENV["TF_VAR_cluster_name"]}/preferences/carla"
-  if ENV["TF_VAR_cluster_name"].start_with?("stg.", "prd.")
-    preferences_url.gsub! "http://", "https://"
+  if ENV["TF_VAR_cluster_name"].start_with?("prd.")
+    wait_for("curl --silent --output /dev/stderr --write-out '%{http_code}' '#{preferences_url}' | grep -q ^2")
+  else
+    wait_for("curl -k --silent --output /dev/stderr --write-out '%{http_code}' '#{preferences_url}' | grep -q ^2")
+    # For staging and dev environments we also need to make sure that certificate is issues by Letsencrypt
+    wait_for("curl -k -vI https://preferences.dev-natarajaya.gpii.net/preferences/carla 2>&1 | grep 'CN=Fake LE Intermediate X1'")
```

Comment:
By default, wait_for will try this command every 20 seconds for 15 minutes. Are those defaults reasonable for this command, especially since if we get to this line then we know the previous command succeeded?

Reply:
Changed to:
```    wait_for(
      "curl -k -vI #{preferences_url} 2>&1 | grep 'CN=Fake LE Intermediate X1'",
      sleep_secs: 5,
      max_wait_secs: 20,
    )```

Reply:
Makes sense